test/networks/views/input_ordering_view.cpp

```diff
@@ -0,0 +1,39 @@
+//
+// Created by benjamin on 11.04.23.
+//
+
+#include <catch2/catch_test_macros.hpp>
+#include "fiction/networks/views/input_ordering_view.hpp"
+#include "fiction/algorithms/network_transformation/fanout_substitution.hpp"
+#include "fiction/networks/technology_network.hpp"
+#include "utils/blueprints/network_blueprints.hpp"
+
+#include <mockturtle/networks/aig.hpp>
+#include <mockturtle/networks/klut.hpp>
+#include <mockturtle/networks/mig.hpp>
+#include <mockturtle/networks/xag.hpp>
+#include <mockturtle/networks/xmg.hpp>
+#include <mockturtle/traits.hpp>
+#include <mockturtle/views/names_view.hpp>
+
+#include <vector>
+
+using namespace fiction;
+
+TEST_CASE("Input_ordering consider all PIs", "[input-ordering]")
+{
+    auto mux21 = blueprints::test_fanout_nodes_rank<mockturtle::names_view<mockturtle::aig_network>>();
+
+    auto ordered_aig_ntk{input_ordering_view{mockturtle::fanout_view{fanout_substitution<mockturtle::names_view<technology_network>>(mux21)}}};
+
+    std::vector<int> nodes_order;
```

Comment:
The node type might be convertible to `int`, but is actually `mockturtle::node<Ntk>` with `Ntk` being your network type.